client/web/src/cody/management/api/hooks/useApiClient.tsx

```diff
@@ -26,62 +27,47 @@ export function useApiCaller<Resp>(call: Call<Resp>): ReactFriendlyApiResponse<R
     const [data, setData] = useState<Resp | undefined>(undefined)
     const [response, setResponse] = useState<Response | undefined>(undefined)
 
-    useEffect(() => {
-        // `ignore` tracks if we should discard any results, because of any underlying race condition
-        // in the sequence of API calls. We return a handle to this in the function callback, which
-        // the React runtime may invoke (setting ignore = true) outside of our view.
-        // https://react.dev/reference/react/useEffect#fetching-data-with-effects
-        // https://maxrozen.com/race-conditions-fetching-data-react-with-useeffect
-        let ignore = false
-
-        ;(async () => {
-            try {
-                const callerResponse = await caller.call(call)
+    const callApi = useCallback(async () => {
+        setLoading(false)
+        try {
+            const callerResponse = await caller.call(call)
 
-                if (ignore) {
-                    return
-                }
-
-                // If we received a 200 response, all is well. We can just return
-                // the unmarshalled JSON response object as-is.
-                setLoading(false)
-                if (callerResponse.response.status >= 200 && callerResponse.response.status <= 299) {
-                    setData(callerResponse.data)
-                    setError(undefined)
-                    setResponse(callerResponse.response)
-                } else {
-                    // For a 4xx or 5xx response this is where we provide any standardized logic for
-                    // error handling. For example:
-                    //
-                    // - On a 401 response, we need to force-logout the user so they can refresh their
-                    //   SAMS access token.
-                    // - On a 500 response, perhaps replace the current UI with a full-page error. e.g.
-                    //   http://github.com/500 or http://github.com/501
-                    setData(undefined)
-                    setError(new Error(`unexpected status code: ${callerResponse.response.status}`))
-                    setResponse(callerResponse.response)
+            // If we received a 200 response, all is well. We can just return
+            // the unmarshalled JSON response object as-is.
+            if (callerResponse.response.status >= 200 && callerResponse.response.status <= 299) {
```

Comment:
I guess this would work here as well:
```suggestion
            if (callerResponse.response.ok) {
```